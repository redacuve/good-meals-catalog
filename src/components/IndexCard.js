import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function IndexCard(props) {
  const {
    color, hasDescription, hasImage, linkTo, meal, str,
  } = props;
  return (
    <Link
      key={meal[str]}
      to={`/${linkTo}/${meal[str]}`}
      className="shadow border rounded bg-white mb-5 cursor-default flex flex-wrap items-center justify-center"
    >
      <div className="w-full md:w-1/2 lg:w-2/3 xl:w-1/2 flex">
        <img
          className="w-full py-1 md:px-1 lg:px-0 xl:px-1"
          src={
            hasImage ? meal.strCategoryThumb : `/assets/img/${meal[str]}.jpg`
          }
          alt={meal[str]}
        />
      </div>
      <div className="w-full md:w-1/2 lg:w-full xl:w-1/2">
        <div className={`w-full bg-${color}-300 text-center`}>
          <h3 className="font-bold text-2xl">{meal[str]}</h3>
        </div>
        {hasDescription && (
          <div className={`w-full bg-${color}-100`}>
            <p className="text-gray-700 text-lg px-3">
              {meal[`${str}Description`]}
            </p>
          </div>
        )}
        <div className="w-full flex justify-center mt-3 mb-1">
          <div
            className={`bg-transparent hover:bg-${color}-500 text-${color}-700 text-sm font-semibold hover:text-white py-2 px-4 border border-${color}-500 hover:border-transparent rounded cursor-pointer`}
          >
            View Category
          </div>
        </div>
      </div>
    </Link>
  );
}

IndexCard.propTypes = {
  color: PropTypes.string.isRequired,
  hasDescription: PropTypes.bool.isRequired,
  hasImage: PropTypes.bool.isRequired,
  linkTo: PropTypes.string.isRequired,
  meal: PropTypes.objectOf(PropTypes.string).isRequired,
  str: PropTypes.string.isRequired,
};

export default IndexCard;
