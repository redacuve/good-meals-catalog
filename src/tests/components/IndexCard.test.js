import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import IndexCard from '../../components/IndexCard';

afterEach(cleanup);
const meal = {
  idCategory: '8',
  strCategory: 'Seafood',
  strCategoryThumb: 'https://www.themealdb.com/images/category/seafood.png',
  strCategoryDescription:
    'Seafood is any form of sea life regarded as food by humans. Seafood prominently includes fish and shellfish. Shellfish include various species of molluscs, crustaceans, and echinoderms. Historically, sea mammals such as whales and dolphins have been consumed as food, though that happens to a lesser extent in modern times. Edible sea plants, such as some seaweeds and microalgae, are widely eaten as seafood around the world, especially in Asia (see the category of sea vegetables). In North America, although not generally in the United Kingdom, the term "seafood" is extended to fresh water organisms eaten by humans, so all edible aquatic life may be referred to as seafood. For the sake of completeness, this article includes all edible aquatic life.',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <IndexCard
        color="indigo"
        hasDescription
        hasImage
        linkTo="category"
        meal={meal}
        str="strCategory"
      />
    </BrowserRouter>,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('render IndexCard correctly', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <IndexCard
        color="indigo"
        hasDescription
        hasImage
        linkTo="category"
        meal={meal}
        str="strCategory"
      />
    </BrowserRouter>,
  );
  expect(getByTestId('meal-title')).toHaveTextContent(
    /^Seafood$/,
  );
});

it('render IndexCard title correctly', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <IndexCard
        color="indigo"
        hasDescription
        hasImage
        linkTo="category"
        meal={meal}
        str="strCategory"
      />
    </BrowserRouter>,
  );
  expect(getByTestId('meal-title')).toContainHTML(
    'Seafood',
  );
});

it('matches snapshot 1', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <IndexCard
          color="indigo"
          hasDescription
          hasImage
          linkTo="category"
          meal={meal}
          str="strCategory"
        />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
