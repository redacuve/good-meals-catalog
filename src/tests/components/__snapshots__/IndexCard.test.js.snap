// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<a
  className="shadow border rounded bg-white mb-5 cursor-default flex flex-wrap items-center justify-center"
  data-testid="single-index-card"
  href="/category/Seafood"
  onClick={[Function]}
>
  <div
    className="w-full md:w-1/2 lg:w-2/3 xl:w-1/2 flex"
  >
    <img
      alt="Seafood"
      className="w-full py-1 md:px-1 lg:px-0 xl:px-1"
      src="https://www.themealdb.com/images/category/seafood.png"
    />
  </div>
  <div
    className="w-full md:w-1/2 lg:w-full xl:w-1/2"
  >
    <div
      className="w-full bg-indigo-300 text-center"
    >
      <h3
        className="font-bold text-2xl"
        data-testid="meal-title"
      >
        Seafood
      </h3>
    </div>
    <div
      className="w-full bg-indigo-100"
    >
      <p
        className="text-gray-700 text-lg px-3"
      >
        Seafood is any form of sea life regarded as food by humans. Seafood prominently includes fish and shellfish. Shellfish include various species of molluscs, crustaceans, and echinoderms. Historically, sea mammals such as whales and dolphins have been consumed as food, though that happens to a lesser extent in modern times. Edible sea plants, such as some seaweeds and microalgae, are widely eaten as seafood around the world, especially in Asia (see the category of sea vegetables). In North America, although not generally in the United Kingdom, the term "seafood" is extended to fresh water organisms eaten by humans, so all edible aquatic life may be referred to as seafood. For the sake of completeness, this article includes all edible aquatic life.
      </p>
    </div>
    <div
      className="w-full flex justify-center mt-3 mb-1"
    >
      <div
        className="bg-transparent hover:bg-indigo-500 text-indigo-700 text-sm font-semibold hover:text-white py-2 px-4 border border-indigo-500 hover:border-transparent rounded cursor-pointer"
      >
        View Category
      </div>
    </div>
  </div>
</a>
`;
